### ISSUE TITLE
[java] UseStringBufferForStringAppends: False positive if only one concatenation

### ISSUE THREAD

**Issue by phoenix384** [pmd/pmd#1736]
**Affects PMD Version:** 6.12

**Rule:** [UseStringBufferForStringAppends](https://pmd.github.io/latest/pmd_rules_java_performance.html#usestringbufferforstringappends)

**Description:**
In the code sample provided below PMD wants me to use a StringBuilder to do the concatenation.
That would be the better solution if there were more than one concatenation, especially inside a loop.
But in this case replacing the only concatenation with a StringBuilder would be the same the compiler does anyway.

**Code Sample demonstrating the issue:**

```java
public void doSomething(String s1) {
	String s2 = "a";
	s2 += s1;
}
```

**Running PMD through:** *Eclipse*

**Comment by oowekyala** [comment]
You're right, this is a false positive. It's also worth noting that javac doesn't convert string concatenation to a StringBuilder when emitting bytecode for release 9 or above ([JEP](https://openjdk.java.net/jeps/280)), so using a StringBuilder would only be useful when loops or collections are involved, and using it for simple stuff like your example is actually less efficient.

I think in fact, we would need a rule that spots useless StringBuilders, which could be replaced with string concatenation. Converting your example to use a stringbuilder would fall into that category.
